src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,179 @@
+package duke.main;
+
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+import duke.exception.InvalidCommandException;
+import duke.exception.OutOfBoundMarkingRequestException;
+import duke.exception.TaskErrorException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+    public static final String botName = "Duke";
+    public static final String listCommand = "list";
+    public static final String byeCommand = "bye";
+
+    private String padding = "       ";
+    private String uselessLine = "-------------------------------------------------------------------------------------";
+    private String addedPhrase = "added: ";
+    private ArrayList<Task> storedItems;
+    private Storage storage;
+    private Parser parser;
+
+    /**
+     * Constructor
+     */
+    public Duke() {
+        // Place-holder constructor, may need to extend later
+        storedItems = new ArrayList<>();
+        storage = new Storage();
+        storage.getDataFile(storedItems);
+        parser = new Parser();
+    }
+
+    /**
+     * Let the bot greet, calls to Ui method
+     */
+    public void greet() {
+        Ui.greet();
+    }
+
+    /**
+     * Takes in the user input, run the Parser to determine what it means and acts accordingly
+     * @param str
+     * @throws InvalidCommandException
+     * @throws OutOfBoundMarkingRequestException
+     * @throws TaskErrorException
+     */
+    public void processUserInput(String str) throws InvalidCommandException, OutOfBoundMarkingRequestException, TaskErrorException {
```

Comment:
Maybe you can keep these methods in another class to make your Duke class looks as clean as possible?